fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -813,6 +820,80 @@ public static void analyzeExpression(Expr expression, AnalyzeState state, Scope
         expressionAnalyzer.analyze(expression, state, scope);
     }
 
+    private Type getReturnTypeOfTruncate(FunctionCallExpr node, List<Type> paramsTypes) {
+        Type firstParamType = paramsTypes.get(0);
+        Expr secondParam = node.getParams().exprs().get(1);
+
+        if (!firstParamType.isDecimalV3()) {
+            throw new IllegalArgumentException("first param of function truncate must be decimalV3");
+        }
+
+        if (!(secondParam instanceof IntLiteral)) {
+            throw new IllegalArgumentException("second param of function truncate must be int literal");
+        }
```

Comment:
Maybe we can't promise the condition, find function will consider type cast, so the type of param may different from the type of function params.

Reply:
This logic has been removed

Reply:
There is the limitation that the second arg of `truncate_decimal` must be int literal or SlotRef expressions.